experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -657,13 +651,16 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
 				maybeWithNode((p) => {
 					// avoid multiple pass side-effects
 					if (!pathVisitors.has(p)) {
-						const visitors: PathVisitor[] = p.events.emitAndCollect(
-							"visitSubtreeChanging",
+						const visitors: (PathVisitor | void)[] = p.events.emitAndCollect(
+							"subtreeChanging",
 							p,
 						);
-						if (visitors.length > 0) pathVisitors.set(p, visitors);
+						if (visitors.length > 0)
+							pathVisitors.set(
+								p,
+								visitors.filter((v): v is PathVisitor => isPathVisitor(v)),
```

Comment:
`void` === `undefined`, so it should be type-safe to write this as `(v): v is PathVisitor => v !== undefined`. Then you would not need to have the type guard function `isPathVisitor`.

Reply:
Done. This is where I started :) but somehow VSCode convinced me that it is insufficient. Now everything looks clean! Started to believe that linting lags or inconsistencies are sometimes present. Learning. Thank you!